aws/resource_aws_appstream_stack.go

```diff
@@ -0,0 +1,570 @@
+		arn := aws.StringValue(resp.Stack.Arn)
+
+		o, n := d.GetChange("tags")
+		if err := keyvaluetags.AppstreamUpdateTags(conn, arn, o, n); err != nil {
+			return diag.FromErr(fmt.Errorf("error updating Appstream Stack tags (%s): %w", d.Id(), err))
+		}
+	}
+
+	return resourceAwsAppStreamStackRead(ctx, d, meta)
+}
+
+func resourceAwsAppStreamStackDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).appstreamconn
+
+	_, err := conn.DeleteStackWithContext(ctx, &appstream.DeleteStackInput{
+		Name: aws.String(d.Id()),
+	})
+	if err != nil {
+		if tfawserr.ErrCodeEquals(err, appstream.ErrCodeResourceNotFoundException) {
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error deleting Appstream Stack (%s): %w", d.Id(), err))
+	}
+
+	// Will wait to finish to delete because after delete it makes a stack inactive then it deletes
+	time.Sleep(waiter.StackSleep)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading Appstream Stack (%s): %w", d.Id(), err))
+	}
+
+	return nil
+}
+
+func expandAccessEndpoints(tfList []interface{}) []*appstream.AccessEndpoint {
+	if len(tfList) == 0 {
+		return nil
+	}
+
+	var apiObjects []*appstream.AccessEndpoint
+
+	for _, v := range tfList {
+		v1 := v.(map[string]interface{})
+
+		endpoint := &appstream.AccessEndpoint{
+			EndpointType: aws.String(v1["endpoint_type"].(string)),
+		}
+		if v2, ok := v1["vpce_id"]; ok {
+			endpoint.VpceId = aws.String(v2.(string))
+		}
+
+		apiObjects = append(apiObjects, endpoint)
+	}
+
+	return apiObjects
+}
+
+func flattenAccessEndpoints(apiObjects []*appstream.AccessEndpoint) []map[string]interface{} {
+	if apiObjects == nil {
+		return nil
+	}
+
+	var tfList []map[string]interface{}
+
+	for _, endpoint := range apiObjects {
+		tfList = append(tfList, map[string]interface{}{
+			"endpoint_type": aws.StringValue(endpoint.EndpointType),
+			"vpce_id":       aws.StringValue(endpoint.VpceId),
+		})
+	}
+
+	return tfList
+}
+
+func expandApplicationSettings(tfList []interface{}) *appstream.ApplicationSettings {
+	if len(tfList) == 0 {
+		return nil
+	}
+
+	apiObject := &appstream.ApplicationSettings{}
+
+	attr := tfList[0].(map[string]interface{})
+	if v, ok := attr["enabled"]; ok {
+		apiObject.Enabled = aws.Bool(v.(bool))
+	}
+	if v, ok := attr["settings_group"]; ok {
+		apiObject.SettingsGroup = aws.String(v.(string))
+	}
+
+	return apiObject
+}
+
+func flattenApplicationSettings(apiObject *appstream.ApplicationSettingsResponse) []interface{} {
+	if apiObject == nil {
+		return nil
+	}
+
+	tfList := map[string]interface{}{}
+	tfList["enabled"] = aws.BoolValue(apiObject.Enabled)
+	tfList["settings_group"] = aws.StringValue(apiObject.SettingsGroup)
+
+	return []interface{}{tfList}
+}
+
+func expandStorageConnectors(tfList []interface{}) []*appstream.StorageConnector {
+	if len(tfList) == 0 {
+		return nil
+	}
+
+	var apiObjects []*appstream.StorageConnector
+
+	for _, v := range tfList {
```

Comment:
Best to break plural flex into 2 functions. https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md#expand-functions-for-blocks